### PR TITLE
Added in settings and configuration

### PR DESCRIPTION
**Pull request by couger01**
- Added v1 of schema
- Added configuration to CSC
- Changed namespace to match csc name from laser to tunablelaser
- Changed doc hierarchy to match standard layout
- Fixed setup.py in preparation for conda recipe

### CODE REVIEW

File: lsst/ts/tunablelaser/csc.py [hunk 1]
```diff
@@ -49,38 +49,37 @@ class LaserErrorCode(enum.IntEnum):
...
-    frequency: float
+    frequency : `float`
         The amount of time that telemetry waits to publish.
     wavelength_topic
     temperature_topic
 
     """
-    def __init__(self, address, configuration, frequency=1, initial_state=salobj.State.STANDBY):
-        super().__init__(SALPY_TunableLaser)
+    def __init__(self,frequency=1, initial_state=salobj.State.STANDBY,index=None,schema="/home/ecoughlin/gitrepo/ts_laser/schema/TunableLaser.yaml"):
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/csc.py]: The default schema is pointing to a file in your user computer. Probably not a good idea. I would suggest you use `schema=None` as the default and, if that is the case, you unpack a relative path.

File: lsst/ts/tunablelaser/csc.py [hunk 2]
```diff
@@ -323,20 +322,51 @@ def begin_disable(self, id_data):
...
+    def configure(self,config):
+        try:
+            self.log.debug(config)
+            self.model.set_configuration(config)
+        except Exception as e:
+            self.fault(code=LaserErrorCode.general_error,report=e.msg)
+            raise
+
+    def get_config_pkg(self):
+        return "ts_config_mttcs"
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/csc.py]: I think the tunable laser will be part of `mtcalsys` not  `mttcs`.

File: lsst/ts/tunablelaser/csc.py [hunk 3]
```diff
@@ -323,20 +322,51 @@ def begin_disable(self, id_data):
                 self.model.stop()
                 self.detailed_state = LaserDetailedState(LaserDetailedState.ENABLEDSTATE)
             except TimeoutError as te:
-                self.fault()
+                self.fault(code=LaserErrorCode.timeout_error,report=te.msg)
+        self.model.set_output_energy_level("OFF")
+
+    def end_start(self,id_data):
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/csc.py]: I think these a coroutines now, as of salobj `v3.12.0`

File: lsst/ts/tunablelaser/csc.py [hunk 4]
```diff
@@ -323,20 +322,51 @@ def begin_disable(self, id_data):
...
+        try:
+            self.model.connect()
+            self.telemetry_task=asyncio.ensure_future(self.telemetry())
+        except Exception as e:
+            raise
+
+    def end_standby(self,id_data):
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/csc.py]: Also coroutine as of salobj v3.12.0

File: lsst/ts/tunablelaser/csc.py [hunk 5]
```diff
@@ -323,20 +322,51 @@ def begin_disable(self, id_data):
...
+        try:
+            if not self.telemetry_task.done():
+                self.telemetry_task.set_result('done')
+            self.model.disconnect()
+        except Exception as e:
+            raise
+
+    def configure(self,config):
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/csc.py]: should be a coroutine

File: lsst/ts/tunablelaser/csc.py [hunk 6]
```diff
@@ -323,20 +322,51 @@ def begin_disable(self, id_data):
...
+        try:
+            self.log.debug(config)
+            self.model.set_configuration(config)
+        except Exception as e:
+            self.fault(code=LaserErrorCode.general_error,report=e.msg)
+            raise
+
+    def get_config_pkg(self):
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/csc.py]: This should be 

```
    @staticmethod
    def get_config_pkg():
    ...
```

File: config.env [hunk 7]
```diff
@@ -0,0 +1 @@
+TS_TUNABLELASER_CONFIG_DIR=~/gitrepo/ts_laser/config
```
Comment (tribeiro) [inline config.env]: What is this file for?

File: lsst/ts/tunablelaser/cli.py [hunk 8]
```diff
@@ -1,31 +1,40 @@
 import argh
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/cli.py]: Can you add some description about this file? Not sure what is it for. I can try to guess it but a doc would be nice!

File: lsst/ts/tunablelaser/settings.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import abc
```
Comment (tribeiro) [inline lsst/ts/tunablelaser/settings.py]: I don't think this classes are used anywhere. Do you really want them here?

File: lsst/ts/tunablelaser/cli.py (hunk 8 above)
Comment (couger01) [reply lsst/ts/tunablelaser/cli.py]: Do you mean the module or the command line interface?

File: lsst/ts/tunablelaser/settings.py (hunk 9 above)
Comment (couger01) [reply lsst/ts/tunablelaser/settings.py]: No.

File: lsst/ts/tunablelaser/cli.py (hunk 8 above)
Comment (tribeiro) [reply lsst/ts/tunablelaser/cli.py]: Whatever you think is best.

File: lsst/ts/tunablelaser/csc.py (hunk 5 above)
Comment (tribeiro) [reply lsst/ts/tunablelaser/csc.py]: This one is still not a coroutine....

Comment (couger01) [reply lsst/ts/tunablelaser/csc.py]: I forgot.

**Pull request merged**